src/nvim/ex_eval.c

```diff
@@ -68,7 +68,16 @@
 #define THROW_ON_INTERRUPT true
 #define THROW_ON_INTERRUPT_TRUE
 
-#define discard_pending_return(p) tv_free((typval_T *)(p))
+static inline void discard_pending_return(void *p)
+  REAL_FATTR_ALWAYS_INLINE REAL_FATTR_PURE;
+
+// Free allocated VimL object and value stored inside
+// @param[in]  p  the pointer to typval_T object
+
+static inline void discard_pending_return(void *p)
```

Comment:
In `src/*.c` files (don't know about test files) just use `FUNC_ATTR_*` attributes on the function definition. No need for the forward declaration (and `REAL_*` attributes) if the function definition is after code like this:
```c
#ifdef INCLUDE_GENERATED_DECLARATIONS
 # include "ex_eval.c.generated.h"
#endif
```
So move the `REAL_*` attributes here and change them into `FUNC_*` attributes.